test/python/circuit/test_circuit_operations.py

```diff
@@ -25,6 +25,19 @@
 class TestCircuitOperations(QiskitTestCase):
     """QuantumCircuit Operations tests."""
 
+    def test_adding_self(self):
+        """Test that qc += qc finishes, which can be prone to infinite while-loops.
+
+        This can occur e.g. when a user tries
+        >>> other_qc = qc
+        >>> other_qc += qc  # or qc2.extend(qc)
+        """
+        qc = QuantumCircuit(1)
+        qc.x(0)  # must contain at least one operation to end up in a infinite while-loop
+
+        # attempt addition, times out if qc is added via reference
+        qc += qc
```

Comment:
While I get that the bug this is fixing is that this will raise an error, it's good practice to at least assert something, even if it's trivial during the execution of a test. Could you add an assert here like `self.assertEqual(['x', 'x'], [x[0].name for x in qc.data])` or something like that

Reply:
Sure, added via 0577b23